cmd/cli/uninstall_mesh.go

```diff
@@ -80,70 +81,300 @@ func newUninstallMeshCmd(config *action.Configuration, in io.Reader, out io.Writ
 				return errors.Errorf("Could not access Kubernetes cluster, check kubeconfig: %s", err)
 			}
 
+			uninstall.extensionsClientset, err = extensionsClientset.NewForConfig(kubeconfig)
+			if err != nil {
+				return errors.Errorf("Could not access extension client set: %s", err)
+			}
+
+			uninstall.meshNamespace = settings.Namespace()
 			return uninstall.run()
 		},
 	}
 
 	f := cmd.Flags()
 	f.StringVar(&uninstall.meshName, "mesh-name", defaultMeshName, "Name of the service mesh")
 	f.BoolVarP(&uninstall.force, "force", "f", false, "Attempt to uninstall the osm control plane instance without prompting for confirmation.  If the control plane with specified mesh name does not exist, do not display a diagnostic message or modify the exit status to reflect an error.")
+	f.BoolVarP(&uninstall.deleteClusterWideResources, "delete-cluster-wide-resources", "a", false, "Cluster wide resources (such as osm CRDs, mutating webhook configurations, validating webhook configurations and osm secrets) are fully deleted from the cluster after control plane components are deleted.")
 	f.BoolVar(&uninstall.deleteNamespace, "delete-namespace", false, "Attempt to delete the namespace after control plane components are deleted")
 	f.Uint16VarP(&uninstall.localPort, "local-port", "p", constants.OSMHTTPServerPort, "Local port to use for port forwarding")
+	f.StringVar(&uninstall.caBundleSecretName, "ca-bundle-secret-name", constants.DefaultCABundleSecretName, "Name of the secret for the OSM CA bundle")
```

Comment:
The CA bundle secret will be removed by the cleanup hook. So wondering why is this required.

Reply:
This is for scenarios when the cleanup hook isn't used